kmod/src/kernelcompat.c

```diff
@@ -0,0 +1,23 @@
+
+#include "kernelcompat.h"
+
+#ifdef KC_SHRINKER_SHRINK
+#include <linux/shrinker.h>
+/*
+ * If a target doesn't have that .{count,scan}_objects() interface then
+ * we have a .shrink() helper that performs the shrink work in terms of
+ * count/scan.
+ */
+int kc_shrink_wrapper(struct shrinker *shrink, struct shrink_control *sc)
+{
+	struct kc_shrinker_funcs *funcs = KC_SHRINKER_FUNCS(shrink);
+	unsigned long nr;
+
+	if (sc->nr_to_scan != 0)
+		funcs->scan_objects(shrink, sc);
```

Comment:
This should get and check the return of scan and translate SHRINK_STOP into -1 and _EMPTY into 0.  (we don't use empty, but we could, and might as well get it right before we forget that there's a compat thing we have to go check if we update a scan function.)